### ISSUE TITLE
False positive: inArray with string and array('value1', 'value2') will always evaluate to true.

### ISSUE THREAD

**Issue by ruudk** [phpstan/phpstan-webmozart-assert#57]
I think that https://github.com/phpstan/phpstan-webmozart-assert/pull/50 is not working as it should.

```php
final class DateFilter
{
    public const DATE_FILTER_TONIGHT      = 'tonight';
    public const DATE_FILTER_TOMORROW     = 'tomorrow';

    public const ALLOWED_FILTERS = [
        self::DATE_FILTER_TONIGHT,
        self::DATE_FILTER_TOMORROW
    ];

    public function __construct(string $dateFilter)
    {
        Assert::inArray(
            $dateFilter,
            self::ALLOWED_FILTERS,
        );
    }
}
```

```
Call to static method Webmozart\Assert\Assert::inArray() with string and array('tonight', 'tomorrow', 'this_weekend', 'this_week', 'this_month', 'next_weekend', 'next_week', 'next_month', ...) will always evaluate to true.
```

/cc @ntzm

**Comment by ruudk** [comment]
Same happens for `oneOf`. I feel it's related to the fact that the array is constructed with constants. Could that be the problem?

**Comment by ruudk** [comment]
No it's not the problem, found another false positive:
```php
$validLocales = [
    'nl_NL', 'en_GB', 'en_US', 'fr_FR', 'be_NL', 'be_FR', 'de_DE', 'sv_SE', 'da_DK', 'es_ES',
];
Assert::oneOf($this->locale->toString(), $validLocales);
```

```
Call to static method Webmozart\Assert\Assert::oneOf() with string and array('nl_NL', 'en_GB', 'en_US', 'fr_FR', 'be_NL', 'be_FR', 'de_DE', 'sv_SE', ...) will always evaluate to true.
```

**Comment by ondrejmirtes** [comment]
This is a duplicate of https://github.com/phpstan/phpstan-webmozart-assert/issues/54, see the explanation.
